civihr_employee_portal/views/includes/views_between_dates_filter_handler.inc

```diff
@@ -60,26 +76,48 @@ class views_between_dates_filter_handler extends date_views_filter_handler_simpl
     // SQL: start date <= value AND end date >= value.
     // If data is correct.
     if (count($field_names) == 2) {
-      $value = $this->get_filter_value('value', $this->value['value']);
+      $value = $this->get_date_value_formatted($this->get_filter_value('value', $this->value['value']));
       $comp_date = new DateObject($value, date_default_timezone(), $this->format);
-
       $start_field = $this->date_handler->sql_field($field_names[0], NULL, $comp_date);
       $start_field = $this->date_handler->sql_format($this->format, $start_field);
       $end_field = $this->date_handler->sql_field($field_names[1], NULL, $comp_date);
       $end_field = $this->date_handler->sql_format($this->format, $end_field);
       $placeholder = $this->placeholder();
       $group = !empty($this->options['date_group']) ? $this->options['date_group'] : $this->options['group'];
       if ($this->operator == '=') {
-        $this->query->add_where_expression($group, "$start_field <= $placeholder AND $end_field >= $placeholder", array($placeholder => $value));
+        if ((int)$this->options['include_null_dates'] === 2) {
+          $this->query->add_where_expression($group, "($start_field <= $placeholder OR $start_field IS NULL) AND ($end_field >= $placeholder OR $end_field IS NULL)", array($placeholder => $value));
+        } else {
+          $this->query->add_where_expression($group, "$start_field <= $placeholder AND $end_field >= $placeholder", array($placeholder => $value));
+        }
       }
       // Else not between
       else {
         $this->query->add_where_expression($group, "$start_field > $placeholder OR $end_field < $placeholder", array($placeholder => $value));
       }
    }
+  }
 
+  /**
+   * Get date value formatted according to date format set up by input field.
```

Comment:
In this case 'get_date_value_formatted()' method returns date by assuming that input date is in 'd/m/Y' format (as it's currently defined in front-end datepicker) (so that's why there is 'explode' stuff). In the other hand, when we don't have datepicker customizing our input format, the most reasonable date format seem to be 'Y-m-d' (for example by using input date filter on View Admin page). Anyway the best solution would be to use the same date format defined for Drupal on front-end and back-end but I have to talk with Bruno and ask him to implement Drupal date format.
Regarding the list() function - it causes PHP notice message if the input array is too short so I wanted to bypass this case.